api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -48,11 +61,30 @@ describe('assessmentsRouter', () => {
   });
 
   describe('DELETE /:assessmentId', () => {
-    it('should delete an assessment in the system', done => {
-      const response = { behaviour: '“Deletes” an assessment in the system' };
+    it('should delete a program assessment in the system if logged-in user is facilitator of that program', done => {
+      const principalId = 3;
+      const assessmentId = 1;
+      mockPrincipalId(principalId);
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([2]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'facilitator' });
+      appAgent.delete(`/${assessmentId}`).expect(204, null, err => {
+        expect(mockGetProgramIdByProgramAssessmentId).toHaveBeenCalledWith(1);
+        expect(mockFindRoleInProgram).toHaveBeenCalledWith(principalId, 2);
+        expect(mockDeleteAssessmentById).toHaveBeenCalledWith(1);
+        done(err);
+      });
+    });
+    it('should return an error if logged-in user is not a facilitator of that program', done => {
+      const principalId = 4;
+      const assessmentId = 1;
+      mockPrincipalId(principalId);
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([2]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'participant' });
       appAgent
-        .delete(`/${exampleAssessmentId}`)
-        .expect(200, itemEnvelope(response), err => {
+        .delete(`/${assessmentId}`)
+        .expect(401, errorEnvelope('Unauthorized user.'), err => {
```

Comment:
Please reference `api/src/middleware/httpErrors.ts` for the exact contents of the error message of an `UnauthorizedError`.